drake/systems/plants/test/testIKMoreConstraints.cpp

```diff
@@ -0,0 +1,221 @@
+#include <drake/RigidBodyIK.h>
+#include <drake/RigidBodyManipulator.h>
+#include <drake/RigidBodyConstraint.h>
+
+#include <drake/IKoptions.h>
+#include <iostream>
+#include <cstdlib>
+#include <limits>
+#include <boost/shared_ptr.hpp>
+
+using namespace std;
+using namespace Eigen;
+
+
+
+int main()
+{
+
+  RigidBodyManipulator rbm("examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator* model = &rbm;
+
+
+  int r_hand, l_foot, r_foot;
+  for(int i = 0;i<model->bodies.size();i++)
+  {
+    if(model->bodies[i]->linkname.compare(string("r_hand")) ==0 )
+    {
+      r_hand = i;
+    }
+    if(model->bodies[i]->linkname.compare(string("l_foot")) ==0 )
+    { 
+      l_foot = i;
+    }
+    if(model->bodies[i]->linkname.compare(string("r_foot")) ==0 )
+    {
+      r_foot = i;
+    }
+  }
+
+
+  if(!model)
+  {
+    cerr<<"ERROR: Failed to load model"<<endl;
+  }
+
+  Vector2d tspan;
+  tspan<<0,1;
+
+  // Default Atlas v5 posture:
+  VectorXd qstar(model->num_positions);
+qstar <<  -0.0260, 0,  0.8440, 0, 0, 0, 0, 0, 0,  0.2700, 0,  0.0550, -0.5700,  1.1300, -0.5500, -0.0550, -1.3300,  2.1530,  0.5000,  0.0985, 0,  0.0008, -0.2700, 0, -0.0550, -0.5700,  1.1300, -0.5500,  0.0550,  1.3300,  2.1530, -0.5000,  0.0985, 0,  0.0008,  0.2564;
+  //model->doKinematics(qstar, v);
+
+  // 1 CoM Constraint
+  Vector3d com_lb = Vector3d::Zero(); 
+  Vector3d com_ub = Vector3d::Zero(); 
+  com_lb(0) = -0.1;
+  com_lb(1) = -0.1;
+  com_lb(2) = 0.6;
+  com_ub(0) = 0.1;
+  com_ub(1) = 0.1;
+  com_ub(2) = 1.0;
+  WorldCoMConstraint* kc_com = new WorldCoMConstraint(model,com_lb,com_ub,tspan);
```

Comment:
Don't use raw pointers. If you must use a pointer, use unique_ptr in this case, but better yet, just use
```
WorldCoMConstraint kc_com(model,com_lb,com_ub,tspan);
```